Remove Autotools support

byrnHDF opened the pull request.

byrnHDF committed:
Remove autotools build support

byrnHDF committed:
Remove autotools build support

byrnHDF committed:
Remove and fix more autotools items

byrnHDF committed:
Remove and fix more autotools items

byrnHDF committed:
Merge branch 'HDFGroup:develop' into develop-remove-at

byrnHDF committed:
Remove more autotool files and references

byrnHDF committed:
Remove more autotool files and references

byrnHDF committed:
Merge with develop

byrnHDF committed:
Merge with develop

byrnHDF committed:
Couple more merges

byrnHDF committed:
Couple more merges

byrnHDF committed:
Fix conflict

byrnHDF committed:
Merge branch 'develop' into develop-remove-at

derobins commented on .gitattributes:
```diff
@@ -30,22 +29,150 @@ config/cmake/scripts/HDF5config.cmake -text
+HDF5Examples/C/H5VDS/h5_vds-percival-unlim-maxmin.c -text
+HDF5Examples/C/H5VDS/h5_vds-percival-unlim.c -text
+HDF5Examples/C/H5VDS/h5_vds-percival.c -text
+HDF5Examples/C/H5VDS/h5_vds-simpleIO.c -text
```
Why do we need these lines for .c and .txt files? Can't git tell that those files are text?

byrnHDF replied:
I wondered but just went ahead and tried to fix what was there.

derobins commented on .gitattributes:
```diff
@@ -350,29 +352,43 @@ java/test/TestH5Tparams.java -text
+java/test/testfiles/JUnit-TestH5Tbasic.txt -text
+java/test/testfiles/JUnit-TestH5Tparams.txt -text
+java/test/testfiles/JUnit-TestUnit.txt -text
+java/test/testfiles/JUnit-TestH5Z.txt -text
```
And why do HDF5 files have the -text attribute (below) ?

byrnHDF replied:
Because they were there before - I guess they were never reviewed

byrnHDF replied:
Removed

derobins commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -17,7 +17,7 @@ A clear and concise description of what you expected to happen.
  - OS and version
  - Compiler and version
- - Build system (e.g. CMake, Autotools) and version
+ - Build system (e.g. CMake) and version
```
This could probably be tweaked. There's only one build system, but I'd be interested in knowing what CMake was targeting: e.g., Xcode vs Makefiles, etc.

byrnHDF replied:
Fixed

derobins commented on HDF5Examples/C/H5D/test-pc.sh:
Why are empty files checked in?

byrnHDF replied:
Not empty on my system?

derobins commented on HDF5Examples/CXX/H5D/expected.out:
Why was this .out file deleted?

byrnHDF replied:
CXX sets - -D "TEST_SKIP_COMPARE=TRUE"

byrnHDF committed:
Fix issues with gitattributes file

byrnHDF committed:
Fix issues with gitattributes file

derobins commented:
I'll go ahead and approve. We can fix stragglers in future PRs.

byrnHDF committed:
remove autotools files from skip list

byrnHDF committed:
remove autotools files from skip list

byrnHDF committed:
Merge branch 'HDFGroup:develop' into develop-remove-at

byrnHDF committed:
Fix codespell issues (#5256)

gheber committed:
Update Dockerfile (#5267)

Added Ninja build tool.

mattjala committed:
Add H5Tdecode2, rename and deprecate H5Tdecode (#5213)

derobins committed:
Simplify the .gitattributes file (#5268)

* Removes all per-file attributes, many of which were incorrectly
  removing the text attribute from text/source files
* Removes all svneol attributes (not sure where those came from, but
  they don't appear in the git attributes docs)
* Marks *.h5 and certain test files as binary
* Removes the !eol attribute, which is unnecessary given text=auto
  and default git text handling

byrnHDF committed:
Updates for updated filters (#5269)

bmribler committed:
Fixed allocation-size-too-big error in H5MM.c (#5076)

A decoded length appeared to be corrupted and had a very large value.
This PR added a check to detect such potential data corruption.

TedLyngmo committed:
Fix a few illegal pointer usages (#5279)

byrnHDF committed:
Correct versions for plugins (#5281)

brtnfld committed:
update the release schedule (#5275)

byrnHDF committed:
Continue an ABI error for java too. (#5282)


* Add the hl-cpp report

haampie committed:
cmake: remove outdated rpath handling (#5271)

jhendersonHDF committed:
Add new -testexpress option to testframe testing framework (#5286)

Adds new -testexpress command-line option to the testframe testing
framework to allow setting or overriding of the TestExpress level
at runtime

Adds macros for the different currently defined TestExpress levels

dependabot[bot] committed:
Bump the github-actions group with 3 updates (#5288)

gheber committed:
Updates/2025 02 04 (#5289)

jhendersonHDF committed:
Minor adjustments to usage of alarm(2) timer in tests (#5292)

Make alarm(2) timer per-test program rather than per-subtest

Avoid enabling alarm(2) timer when TestExpress is set to 0

jhendersonHDF committed:
Fix segfault in H5Scombine_select (#5296)

Checks for a hyperslab selection in the resulting dataspace before
attempting to set a hyperslab-specific field

fortnern committed:
Fix error in H5Ddebug() (#5299)

* Fix error in H5Ddebug(). Add h5ls test cases.

* Committing clang-format changes

* Fix cmake (hopefully)

* Fix cmake again (hopefully)

* Fix issue with modification times showing up differently on different
systems. Add test case for v1 b-trees.

* Add cmake for new tests

* Add RELEASE.txt entry

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

gheber committed:
Add pkg-config. (#5304)

byrnHDF committed:
Update files based on issues discovered during release (#5294)

byrnHDF committed:
Fix conflict

byrnHDF committed:
Overwrite

byrnHDF commented:
Replaced by:  Remove Autotools support #5308
